File: src/commonMain/kotlin/com/scoreo/application/UpdateGameTypeUseCase.kt
package com.scoreo.application

import com.scoreo.domain.model.GameType
import com.scoreo.domain.port.GameTypeRepository

class UpdateGameTypeUseCase(private val repository: GameTypeRepository) {
    operator fun invoke(gameType: GameType) {
        repository.save(gameType)
    }
}

File: apps/scoreo/src/application/updateGameTypeUseCase.ts
import type { GameType } from '../domain/model/gameType'
import type { GameTypeRepository } from '../domain/port/gameTypeRepository'

export class UpdateGameTypeUseCase {
  constructor(private readonly repository: GameTypeRepository) {}

  invoke(gameType: GameType): void {
    this.repository.save(gameType)
  }
}
